src/main/java/org/joshsim/lang/interpret/fragment/ProgramBuilder.java

```diff
@@ -45,7 +45,7 @@ public ProgramBuilder() {
    * @param fragment The fragment to add to the program
    * @throws IllegalArgumentException if the fragment type is not supported at the top level
    */
-  public void add(Fragment fragment) {
+  public void add(org.joshsim.lang.interpret.fragment.josh.JoshFragment fragment) {
```

Comment:
Let's use `JoshFragment` with an import instead of fully qualified name `org.joshsim.lang.interpret.fragment.josh.JoshFragment`.